### ISSUE TITLE
spark2.4 v0.23.2-preview has a fat jar issue packing org.bouncycastle 

### ISSUE THREAD

**Issue by yeshvantbhavnasi** [GoogleCloudDataproc/spark-bigquery-connector#571]
v0.23.2 verison fat jar have the bouncycastle classes on the classpath. not on the repackaed folders seems to be a bug on the packaging

> packages contents on extract
```
spark-2.4-bigquery-0.23.2-preview
   - org 
       - bouncycastle
          - crypto
              - CryptoServicesRegistrar.class
```

Causing issues on build if the project already have bc-fips dependency. We might have to fix this. v0.23.2 prior version doesnt have this problem.

**Comment by yeshvantbhavnasi** [comment]
![Screen Shot 2022-03-23 at 4 31 15 PM](https://user-images.githubusercontent.com/20260478/159814765-cc509554-c9b0-4c74-9132-5ed25390021b.png)

**Comment by davidrabinowitz** [comment]
Fixed in version 0.24.0
